scripts/coverage.sh

```diff
@@ -1,37 +1,29 @@
 #!/bin/sh
-# Runs tests and collects code coverage to ./cov.info file
+# Runs tests and collects code coverage to ./lcov.info file
 
 set -e # aborts as soon as anything returns non-zero exit status
 
 
 PROJECT_DIR=$(pwd -P)
 
-COVERAGE_FILE="$PROJECT_DIR"/cov.info
-INSTRUMENTED_CODE_DIR="$PROJECT_DIR"/src-cov
+COVERRAGE_DIR="$PROJECT_DIR"/coverage
+COVERAGE_FILE="$PROJECT_DIR"/coverage/lcov.info
 BIN_DIR="$PROJECT_DIR"/node_modules/.bin
+TEST_GLOB=dist/test/**/*-test.coffee
 
 
 # Cleanup & preparation
-rm -rf "$INSTRUMENTED_CODE_DIR" "$COVERAGE_FILE"
-mkdir "$INSTRUMENTED_CODE_DIR"
-
-
-# Creating directory with instrumented JS code
-"$BIN_DIR"/coffeeCoverage \
-    --exclude=node_modules,.git,test,scripts \
-    --path=relative \
-    "$PROJECT_DIR" "$INSTRUMENTED_CODE_DIR" 1>&2
-
-cp "$PROJECT_DIR"/package.json "$INSTRUMENTED_CODE_DIR"
-cp -r "$PROJECT_DIR"/test "$INSTRUMENTED_CODE_DIR"/test
+rm -rf "$COVERRAGE_DIR"
+rm -rf dist
+npm run copy:test
+npm run build
 
 
 # Testing
 export COVERAGE_FILE
-cd "$INSTRUMENTED_CODE_DIR" && npm test
+istanbul cover ./node_modules/.bin/_mocha -- \
```

Comment:
What is `_mocha` with leading underscore?

Reply:
Not using `npm test` here means we need to keep two different versions of the test command. And not using the same `--opts` as the `npm test` means we're running a different test setup on CI and a different test setup locally.

If this is needed for the part where tests are still in CoffeeScript, then OK. In the next iteration I'd like to have just one testing command specified in one place, which is used when doing both `npm test` and `npm test:coverage`.

Reply:
Also, I'm wondering whether in the next step we could simplify the coverage scripts to a level when it would be feasible to rewrite them in JS (because https://github.com/apiaryio/dredd/issues/763)

Reply:
`_mocha` is Mocha's "real" executable. From `mocha` in `node_modules/bin` - `This tiny wrapper file checks for known node flags and appends them when found, before invoking the "real" _mocha(1) executable.`. I read in `istanbul`'s GH issues that using `_mocha` solves a problem with non-working coverage when using ES6+ for some. It works correctly with `mocha` for us though - I'll change it back to `mocha`.